### ISSUE TITLE
Progress bar and current % of download (useful for big files)

### ISSUE THREAD

**Issue by jfoclpf** [kevva/download#227]
Any tips on how to use progress bar or % for the download (very useful for big files)?

**Comment by jfoclpf** [comment]
Created PR https://github.com/kevva/download/pull/228 which addresses the current issue

**Comment by jfoclpf** [comment]
For big files, it may be useful to use a progress bar. You must install also module `progress`.

```js
const download = require('download')
const ProgressBar = require('progress')

const writeStream = fs.createWriteStream('http://unicorn.com/foo.zip')
const readStream = download('dist/foo.zip')
readStream.on('response', function (res) {
  const len = parseInt(res.headers['content-length'], 10)
  const bar = new ProgressBar('  downloading [:bar] :rate/bps :percent :etas', {
    complete: '=',
    incomplete: ' ',
    width: 20,
    total: len
  })
  readStream.on('data', function (chunk) {
    writeStream.write(chunk)
    bar.tick(chunk.length)
  })
  readStream.on('end', function () {
    console.log('Download done with success\n')
    writeStream.end()
  })
})
```

![image](https://user-images.githubusercontent.com/3984909/185933802-8b448e9c-cf82-4ce4-8684-d49be2d7e77d.png)

**Comment by codad5** [comment]
> For big files, it may be useful to use a progress bar. You must install also module `progress`.
> 
> ```js
> const download = require('download')
> const ProgressBar = require('progress')
> 
> const writeStream = fs.createWriteStream('http://unicorn.com/foo.zip')
> const readStream = download('dist/foo.zip')
> readStream.on('response', function (res) {
>   const len = parseInt(res.headers['content-length'], 10)
>   const bar = new ProgressBar('  downloading [:bar] :rate/bps :percent :etas', {
>     complete: '=',
>     incomplete: ' ',
>     width: 20,
>     total: len
>   })
>   readStream.on('data', function (chunk) {
>     writeStream.write(chunk)
>     bar.tick(chunk.length)
>   })
>   readStream.on('end', function () {
>     console.log('Download done with success\n')
>     writeStream.end()
>   })
> })
> ```
> 
> ![image](https://user-images.githubusercontent.com/3984909/185933802-8b448e9c-cf82-4ce4-8684-d49be2d7e77d.png)

Isn't the create write stream to be the download path while the first parameter in the download function is the URL?

**Comment by jfoclpf** [comment]
No, it might seem strange but I am almost sure the code is correct, I tested myself.

**Comment by codad5** [comment]
I had to switch it to work fine tho .
